Add time picker for Windows

rectified95 opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please follow the template so that the reviewers can easily understand what the code changes affect -->

# Summary
Implemented the 'time' mode for Windows. Now it's on par with iOS and Android.
Implemented the following properties: is24Hour, minuteInterval.
Added sample usage to the Example app. Above props can be toggled using a dropdown box.

Closes microsoft/react-native-windows#4695

<!--
Explain the **motivation** for making this change: here are some points to help you:

* What issues does the pull request solve? Please tag them so that they will get automatically closed once the PR is merged
* What is the feature? (if applicable)
* How did you implement the solution?
* What areas of the library does it impact?
-->
[
![windows_time_1](https://user-images.githubusercontent.com/12816515/86052712-43e9ef80-ba0c-11ea-8229-a28e94b327cf.png)
](url)
![windows_time_2](https://user-images.githubusercontent.com/12816515/86052739-4f3d1b00-ba0c-11ea-8dc5-3a8a8f87440a.png)


## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Windows |    ✅     |

## Checklist

<!-- Check completed item, when applicable, via: [X] -->

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`)

asklar commented on windows/DateTimePickerWindows/DateTimePickerView.idl:
```diff
@@ -8,4 +8,10 @@ namespace DateTimePicker {
     };
+
+    [default_interface]
+    runtimeclass TimePickerView : Windows.UI.Xaml.Controls.TimePicker {
```
FYI once we'll start making modules work with WinUI 3 we'll need to change these things to inherit from `XAML_NAMESPACE.Controls.TimePicker` etc.

asklar commented on windows/DateTimePickerWindows/TimePickerView.cpp:
```diff
@@ -0,0 +1,109 @@
+#include <winrt/Windows.Globalization.h>
+
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+}
```
is this actually being used?

rectified95 replied:
Yes, Microsoft::ReactNative is used in winrt::IReactContext;
Windows::Foundation is used in winrt::TimeSpan

rectified95 replied:
Is there a reason we want to minimize the usage of winrt namespace aliases? I remember this being an issue in another PR.

asklar replied:
it obscures where types/apis come from. Also especially relevant when the same name is used for types in different namespaces (e.g. `Windows.UI.ApplicationModel.Activation.LaunchActivationEventArgs` versus `Microsoft.UI.ApplicationModel.Activation.LaunchActivationEventArgs`)

asklar commented on windows/DateTimePickerWindows/TimePickerView.cpp:
```diff
@@ -0,0 +1,109 @@
+
+            auto nowInSeconds = winrt::clock::to_time_t(winrt::DateTime::clock::now());
+            auto daysSinceDawnOfTime = nowInSeconds / secondsPerDay;
+            auto tickCount = (daysSinceDawnOfTime * secondsPerDay) * 1000 + selectedTime;
```
watch for overflows?

rectified95 commented:
@vonovak Can you review this one?

vonovak commented:
@rectified95 please assign me as a reviewer when you're done here, thanks!

rectified95 commented:
@vonovak I don't have access right to do anything here :D That includes adding reviewers

vonovak commented:
@rectified95 you have a pending invite to this repository since you did the last PR, can you please take a look and I'll resend it in case you don't have it. Thanks!

vonovak commented on src/datetimepicker.windows.js:
```diff
@@ -50,13 +51,26 @@ export default class DateTimePickerWindows extends React.Component<WindowsNative
-      />
-    );
+    // 'date' is the default mode
+    if (this.props.mode === 'date' || this.props.mode === undefined) {
```
would you please rewrite this to use function component (with hooks) and also consider the case when `mode === null` (should be handled the same as `undefined` I think). Thanks!

rectified95 replied:
Rewrote as function component. 
Handled both undefined and null using the following comparison:
"mode **==** null"

rectified95 commented:
@vonovak Sorry for the late reply - I've been off this week. I'll rewrite the component using hooks on Monday.

About the invite: it got lost in my inbox, sorry about that. Now it says it's expired - can you, please, resend it? Thanks!

rectified95 commented:
@vonovak 
Addressed your remarks.
Per my previous comment - kindly resend the collab invite for this repo - the previous one has already expired.
Thanks!

vonovak commented:
@rectified95 I have re-sent the invite, can you please accept it? Thanks!

rectified95 commented:
@vonovak Accepted, thank you!
